src/shared/extensions/git.ts

```diff
@@ -0,0 +1,124 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { execFile } from 'child_process'
+import { promisify } from 'util'
+import * as vscode from 'vscode'
+import * as GitTypes from '../../../types/git.d'
+import { VSCODE_EXTENSION_ID } from '../extensions'
+import { getLogger } from '../logger/logger'
+
+/**
+ * Wrapper around the internal VS Code Git extension.
+ *
+ * The Git extension is a special case in that its API has an additional 'enablement' mechanism where-in
+ * the extension may be active, but its features have intentionally been disabled by the user. The current
+ * implementation 'silenty' (it still logs something) fails when the API is disabled since the common case
+ * uses these methods supplementary to other functionality rather than as a dependency. A caller can always
+ * check the `enabled` field to decisively determine if the extension is active.
+ */
+export class GitExtension {
+    private api?: GitTypes.API
+
+    public get enabled(): boolean {
+        return this.api !== undefined
+    }
+
+    constructor() {
+        this.init()
+    }
+
+    private init(): void {
+        const ext = vscode.extensions.getExtension<GitTypes.GitExtension>(VSCODE_EXTENSION_ID.git)
+
+        if (ext === undefined) {
+            getLogger().info(
+                `The "${VSCODE_EXTENSION_ID.git}" extension was not found. Git related features will be disabled.`
+            )
+            return
+        }
+
+        const whenActive = () => {
+            this.api = ext.exports.enabled ? ext.exports.getAPI(1) : undefined
+            ext.exports.onDidChangeEnablement(enabled => {
+                this.api = enabled ? ext.exports.getAPI(1) : undefined
```

Comment:
would be nice to make a generic wrapper to deal with this, but I see that onDidChangeEnablement is specific to the Git extension. If it's a common pattern we could feature-detect that symbol.

Reply:
Yup :/
If there was a generic `onDidEnable`/`onDidActivate` hook for extensions it would solve many issues.